homeassistant/components/mqtt/mixins.py

```diff
@@ -210,6 +217,14 @@ async def async_discover(discovery_payload):
     )
 
 
+def init_entity_id_from_config(hass, entity, config, entity_id_format):
+    """Set entity_id from object_id if defined in config."""
+    if CONF_OBJECT_ID in config:
+        entity.entity_id = generate_entity_id(
```

Comment:
`generate_entity_id()` is deprecated, we should call `async_generate_entity_id()`
```suggestion
        entity.entity_id = async_generate_entity_id(
```

Reply:
done